File: backend/app/core/security.py
from datetime import datetime,timedelta,timezone
from jose import jwt
from passlib.context import CryptContext
from app.core.config import get_settings
pwd_context=CryptContext(schemes=['bcrypt'],deprecated='auto')
ALGORITHM='HS256'
def hash_password(password:str)->str:return pwd_context.hash(password)
def verify_password(plain_password:str,password_hash:str)->bool:return pwd_context.verify(plain_password,password_hash)
def create_access_token(subject:str,expires_minutes:int|None=None)->str:
    s=get_settings(); exp=datetime.now(timezone.utc)+timedelta(minutes=expires_minutes or s.access_token_expire_minutes)
    return jwt.encode({'sub':subject,'exp':exp},s.jwt_secret_key,algorithm=ALGORITHM)

File: app/Core/Security.php
<?php

declare(strict_types=1);

namespace App\Core;

final class Security
{
    public static function hashPassword(string $password): string
    {
        return password_hash($password, PASSWORD_DEFAULT);
    }

    public static function verifyPassword(string $password, string $hash): bool
    {
        return password_verify($password, $hash);
    }

    public static function createAccessToken(string $subject, ?int $expiresMinutes = null): string
    {
        $minutes = $expiresMinutes ?? (int) Env::get('ACCESS_TOKEN_EXPIRE_MINUTES', 60);
        $secret = (string) Env::get('JWT_SECRET_KEY', 'change_me');
        $header = ['alg' => 'HS256', 'typ' => 'JWT'];
        $payload = ['sub' => $subject, 'exp' => time() + ($minutes * 60)];

        $segments = [
            self::base64UrlEncode((string) json_encode($header, JSON_UNESCAPED_SLASHES)),
            self::base64UrlEncode((string) json_encode($payload, JSON_UNESCAPED_SLASHES)),
        ];
        $signature = hash_hmac('sha256', implode('.', $segments), $secret, true);
        $segments[] = self::base64UrlEncode($signature);

        return implode('.', $segments);
    }

    private static function base64UrlEncode(string $value): string
    {
        return rtrim(strtr(base64_encode($value), '+/', '-_'), '=');
    }
}
